google.com/src/ada_google_com/sheets.py

```diff
@@ -1,133 +1,239 @@
+        """Factory method to construct object from header"""
+        kwarg = {
+            k: cls._parse(
+                field_name=k,
+                head=head,
+                accepted_values=getattr(cls.Fields, k.upper()).value,
+                sheet_name=sheet_name,
+            ) for k in ('name', 'user', 'email', 'last_time_logged_in')
+        }
+
+        return cls(sheet_name=sheet_name, **kwarg)
+
+    @staticmethod
+    def _parse(field_name: str,
+               head: Sequence[str],
+               accepted_values: Sequence[str],
+               sheet_name: str) -> int:
+        """Parse header to extract name, user, email, last_time_logged_in"""
+        for name in accepted_values:
+            if name in head:
+                return head.index(name)
+
+        raise ValueError(f'No {field_name!r} {accepted_values!r} field to be processed in sheet '
+                         f'{sheet_name!r}')
+
+
+class Foot(BaseModel):
+    """
+    Wrapper for last row in spreadsheet
+
+    Example::
 
+        ------------------------------------------------------------------------
+        | Name      | User | Email        | Last time logged In                |
+        | gmail.com | A1   | a1@gmail.com | 2022.12.28                         |
+        |           |       |             | notify if when: more than 180 days |
+        ------------------------------------------------------------------------
 
-def process_head_content_and_footer(  # pylint: disable=too-many-locals
-        head: Sequence[str],
-        content: Sequence[Sequence[str]],
-        footer: Sequence[str],
-        sheet_name: str,
-        output: List[str],
-    ) -> None:
     """
-    Process data available in head, content, and footer to send notification if needed
+    notify_after_n_days: int
+
+    @classmethod
+    def from_sheet_foot_and_head_index(cls, foot: Sequence[str], head_index: HeadIndex) -> 'Foot':
+        """Factory method to construct object from raw footer and HeadIndex"""
+        notify_after_n_days = cls._extract_notify_after_n_days_value(
+            foot=foot,
+            head_index=head_index
+        )
+
+        return cls(notify_after_n_days=notify_after_n_days)
+
+    @staticmethod
+    def _extract_notify_after_n_days_value(foot: Sequence[str], head_index: HeadIndex) -> int:
+        try:
+            notify_after_n_days_raw_value = foot[head_index.last_time_logged_in]
+        except IndexError as index_error:
+            raise IndexError(
+                f'Footer {foot!r} does not contain '
+                f'{head_index.Fields.LAST_TIME_LOGGED_IN.value!r} field'
+            ) from index_error
+
+        regexp = re.compile(
+            r'Notify if when: more than (\d+) days',
+            re.IGNORECASE
+        )
+        match = regexp.search(notify_after_n_days_raw_value)
+        if not match:
+            print(f'Can not match {regexp.pattern}')
+            sys.exit(1)
```

Comment:
Why using such rough syntax? Can't a proper exception be raised there?

Reply:
fixed